datafusion/physical-plan/src/repartition/mod.rs

```diff
@@ -611,27 +607,49 @@ impl RepartitionExec {
                 abort_helper: Arc::new(AbortOnDropMany::<()>(vec![])),
             })),
             metrics: ExecutionPlanMetricsSet::new(),
-            preserve_order: false,
+            preserve_order,
             cache,
+        })
+    }
+
+    fn maintains_input_order_helper(
+        input: &Arc<dyn ExecutionPlan>,
+        preserve_order: bool,
+    ) -> Vec<bool> {
+        if preserve_order {
+            vec![true]
+        } else {
+            // We preserve ordering when input partitioning is 1
+            vec![input.output_partitioning().partition_count() <= 1]
         }
```

Comment:
Why not write it as `vec![preserve_order || input.output_partitioning().partition_count() <= 1]`